cli/cmdlineparser.cpp

```diff
@@ -544,6 +544,10 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
                 }
             }
 
+            else if (std::strcmp(argv[i], "--cpp-header-probe") == 0) {
```

Comment:
It's not just for C++ markers right it's also for C markers? I wonder if "cpp-" should be removed from the flag.

Reply:
We treat headers as C by default. C headers always have an extension. IIRC all files in `/usr/include/` which had this marker only used it to specify `C++`.